File: No_3016/Minimum Number of Pushes to Type Word II.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        counter = [0] * 26
        for c in word:
            counter[ord(c) - ord('a')] += 1
            
        counter.sort(reverse=True)
        
        ret = 0
        char_num = 0
        KEY_NUM = 8
        
        for count in counter:
            ret += ((int)(char_num / KEY_NUM) + 1) * count
            char_num += 1
        
        return ret

File: No_3016/Minimum Number of Pushes to Type Word II.cpp
class Solution {
public:
    int minimumPushes(string word) {
        vector<int> counter(26);
        for (char c : word)
            counter[c - 'a']++;
        
        sort(counter.begin(), counter.end(), greater<int>());
        
        int ret = 0, charNum = 0, KEYNUM = 8;
        for (int count : counter) {
            if (count != 0) {
                ret += ((charNum / KEYNUM) + 1) * count;
                charNum++;
            }
        }
        
        return ret;
    }
};